scripts/deploy/src/install.js

```diff
@@ -0,0 +1,136 @@
+import fs from 'fs';
+import child_process from 'child_process';
+import fetch from 'node-fetch';
+import yaml from 'js-yaml';
+import path from 'path';
+
+const MEDIC_REPO_NAME = 'medic';
+const MEDIC_REPO_URL = 'https://docs.communityhealthtoolkit.org/helm-charts';
+const CHT_CHART_NAME = `${MEDIC_REPO_NAME}/cht-chart-4x`;
+const DEFAULT_CHART_VERSION = '1.0.*';
+
+import { fileURLToPath } from 'url';
+import { obtainCertificateAndKey, createSecret } from './certificate.js';
+import UserRuntimeError from './error.js';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const readFile = function(f) {
+  try {
+    return yaml.load(fs.readFileSync(f, 'utf8'));
+  } catch (err) {
+    console.error(err);
+    process.exit(1);
+  }
+};
+
+const prepare = function(f) {
+  const values = readFile(f);
+  const environment = values.environment || '';
+  const scriptPath = path.join(__dirname, 'prepare.sh');
+  child_process.execSync(`${scriptPath} ${environment}`, { stdio: 'inherit' }); //NoSONAR
+};
+
+const loadValues = function(f) {
+  if (!f) {
+    console.error('No values file provided. Please specify a values file using -f <file>');
+    process.exit(1);
+  }
+  return readFile(f);
+};
+
+const determineNamespace = function(values) {
+  const namespace = values.namespace || '';
+  if (!namespace) {
+    console.error('Namespace is not specified.');
+    process.exit(1);
+  }
+  return namespace;
+};
+
+const get_image_tag = async function(chtversion) {
+  const response = await fetch(`https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:${chtversion}`);
+  const data = await response.json();
+  const tag = data.tags && data.tags.find(tag => tag.container_name === 'cht-api');
+  if (!tag) {
+    return Promise.reject(new UserRuntimeError('cht image tag not found'));
+  }
+  return tag.image.split(':').pop();
+};
+
+const get_chart_version = function(values) {
+  return values.cht_chart_version || DEFAULT_CHART_VERSION;
+};
+
+const helmInstallOrUpdate = function(valuesFile, namespace, values, image_tag) { //NoSONAR
+  const chart_version = get_chart_version(values);
+  ensureMedicHelmRepo();
+  const project_name = values.project_name || '';
+  const namespaceExists = checkNamespaceExists(namespace);
+
+  try {
+    const releaseExists = child_process.execSync(`helm list -n ${namespace}`).toString(); //NoSONAR
+    if (releaseExists.includes(project_name)) { //NoSONAR
+      console.log('Release exists. Performing upgrade.');
+      child_process.execSync(`helm upgrade` + //NoSONAR
```

Comment:
I think it would be much cleaner to have a wrapper that runs helm commands, so that it's more visible which parameters we are passing. something like: 

```
const helmCmd = (action, params) => {
  const paramsArray = Object.entries(params).map(([ key, value ]) => {
    if (!value) {
      return `--${key}`;
    }
    return `--${key} ${value}`
  });
  return child_process.execSync(`helm ${action} ${paramsArray.join(' ')}`);
}

const opts = {
  install: `${projectName} ${CHT_CHART_NAME}`,
  version: chartVersion,
  namespace,
  set: `cht_image_tag=${image_tag}`,
};
helmCmd('install', opts);
```

Reply:
Thanks. I like this. Changed.